flocker/provision/_digitalocean.py

```diff
@@ -0,0 +1,370 @@
+
+    # Import here, so that this can be added to ``flocker.provision`` without
+    # having to install ``pyocean``.
+    import pyocean
+
+    v2client = pyocean.DigitalOcean(access_token=token)
+    v2droplet = v2client.droplet.get(node._node.id)
+
+    set_droplet_kernel(v2droplet, DIGITALOCEAN_KERNEL)
+    run(
+        username='root',
+        address=node.address,
+        commands=task_install_digitalocean_kernel()
+    )
+
+    # Libcloud doesn't support shutting down DO vms.
+    # See https://issues.apache.org/jira/browse/LIBCLOUD-655
+    retry_if_pending(v2droplet.shutdown)
+
+    # Libcloud doesn't support powering up DO vms.
+    # See https://issues.apache.org/jira/browse/LIBCLOUD-655
+    # Even after the shutdown, the droplet may not be quite ready to power on,
+    # so also check for that resulting error here.
+    retry_on_error([pending_event, droplet_still_on], v2droplet.power_on)
+
+    # Finally run all the standard Fedora20 installation steps.
+    run(
+        username='root',
+        address=node.address,
+        commands=provision(
+            package_source=package_source,
+            distribution=node.distribution,
+        )
+    )
+    return node.address
+
+
+IMAGE_NAMES = {
+    # It'd be better to use image ID here, but the following code is currently
+    # written to lookup image names...which would normally be good for
+    # readability but which in the case DigitalOcean are pretty meaningless.
+    'fedora-20': '20 x64',
+}
+
+
+def location_by_slug(driver, location_slug):
+    """
+    Look up a DigitalOcean location by its short human readable "slug" code.
+
+    # XXX: ``libcloud.DigitalOceanDriver.list_locations`` discards the slug
+    # so we make a direct call to the v1 API and parse the returned dictionary.
+    # See https://issues.apache.org/jira/browse/LIBCLOUD-653
+
+    :param driver: The libcloud driver to query for sizes.
+    :param bytes location_slug: A DigitalOcean location "slug".
+    :returns: ``NodeLocation``.
+    """
+    result = driver.connection.request('/regions')
+    for location_dict in result.object['regions']:
+        if location_dict['slug'] == location_slug:
+            break
+    else:
+        raise ValueError("Unknown location slug.", location_slug)
+
+    return driver._to_location(location_dict)
+
+
+def size_by_slug(driver, size_slug):
+    """
+    Look up a DigitalOcean size by its short human readable "slug" code.
+
+    # XXX: ``libcloud.DigitalOceanDriver.list_sizes`` discards the slug
+    # so we make a direct call to the v1 API and parse the returned dictionary.
+    # See https://issues.apache.org/jira/browse/LIBCLOUD-654
+
+    :param driver: The libcloud driver to query for sizes.
+    :param bytes size_slug: A DigitalOcean size "slug".
+    :returns: ``NodeSize``.
+    """
+    result = driver.connection.request('/sizes')
+    for size_dict in result.object['sizes']:
+        if size_dict['slug'] == size_slug:
+            break
```

Comment:
:question: This would be more readable if `return driver._to_size(size_dict)` too the place of break and then you don't need an `else`.